modules/ismp/pallet/pallet-asset-transfer/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+// Copyright (C) Polytope Labs Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod xcm_utilities;
+
+extern crate alloc;
+
+use alloc::{boxed::Box, string::ToString};
+use core::marker::PhantomData;
+
+use alloy_rlp_derive::{RlpDecodable, RlpEncodable};
+use frame_support::{
+    ensure,
+    traits::{fungibles, Get},
+};
+use ismp::{
+    events::Meta,
+    host::StateMachine,
+    module::IsmpModule,
+    router::{DispatchPost, DispatchRequest, IsmpDispatcher, Request, Timeout},
+    util::hash_request,
+};
+pub use pallet::*;
+use pallet_ismp::{dispatcher::Dispatcher, host::Host};
+use sp_core::{H160, H256, U256};
+use sp_runtime::traits::AccountIdConversion;
+use staging_xcm::{
+    v3::{AssetId, Fungibility, Junction, MultiAsset, MultiAssets, MultiLocation, WeightLimit},
+    VersionedMultiAssets, VersionedMultiLocation,
+};
+use xcm_utilities::MultiAccount;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use alloc::vec;
+    use frame_support::{pallet_prelude::*, traits::fungibles, PalletId, Parameter};
+    use sp_runtime::Percent;
+
+    #[pallet::pallet]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    /// The config trait
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_ismp::Config + pallet_xcm::Config {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// The asset tranfer's pallet id, used for deriving its sovereign account ID.
+        /// All escrowed funds will be custodied by this account
+        #[pallet::constant]
+        type PalletId: Get<PalletId>;
+
+        /// Protocol fees will be custodied by this account
+        #[pallet::constant]
+        type ProtocolAccount: Get<PalletId>;
```

Comment:
why not just use the `PalletId` for this?

Reply:
Wanted to have a separate account that custodies protocol fees, just for ease of tracking

Reply:
> Wanted to have a separate account that custodies protocol fees, just for ease of tracking

But the `PalletId` will not be used for any other fees?

Reply:
If we use the same account, it might be more tricky to seperate  protocol fees from TVL if we change the protocol fee percentage a couple times.